store/sqlstore/upgrade.go

```diff
@@ -1334,27 +1336,114 @@ func hasMissingMigrationsVersion536(sqlStore *SqlStore) bool {
 }
 
 func upgradeDatabaseToVersion537(sqlStore *SqlStore) {
-	// if shouldPerformUpgrade(sqlStore, Version5360, Version5370) {
-	sqlStore.RemoveIndexIfExists("idx_posts_channel_id", "Posts")
-	sqlStore.RemoveIndexIfExists("idx_channels_name", "Channels")
-	sqlStore.RemoveIndexIfExists("idx_publicchannels_name", "PublicChannels")
-	sqlStore.RemoveIndexIfExists("idx_channelmembers_channel_id", "ChannelMembers")
-	sqlStore.RemoveIndexIfExists("idx_emoji_name", "Emoji")
-	sqlStore.RemoveIndexIfExists("idx_oauthaccessdata_client_id", "OAuthAccessData")
-	sqlStore.RemoveIndexIfExists("idx_oauthauthdata_client_id", "OAuthAuthData")
-	sqlStore.RemoveIndexIfExists("idx_preferences_user_id", "Preferences")
-	sqlStore.RemoveIndexIfExists("idx_notice_views_user_id", "ProductNoticeViewState")
-	sqlStore.RemoveIndexIfExists("idx_notice_views_user_notice", "ProductNoticeViewState")
-	sqlStore.RemoveIndexIfExists("idx_status_user_id", "Status")
-	sqlStore.RemoveIndexIfExists("idx_teammembers_team_id", "TeamMembers")
-	sqlStore.RemoveIndexIfExists("idx_teams_name", "Teams")
-	sqlStore.RemoveIndexIfExists("idx_user_access_tokens_token", "UserAccessTokens")
-	sqlStore.RemoveIndexIfExists("idx_user_terms_of_service_user_id", "UserTermsOfService")
-	sqlStore.RemoveIndexIfExists("idx_users_email", "Users")
-	sqlStore.RemoveIndexIfExists("idx_sharedchannelusers_user_id", "SharedChannelUsers")
-	sqlStore.RemoveIndexIfExists("IDX_RetentionPolicies_DisplayName_Id", "RetentionPolicies")
-	sqlStore.CreateIndexIfNotExists("IDX_RetentionPolicies_DisplayName", "RetentionPolicies", "DisplayName")
-
-	// saveSchemaVersion(sqlStore, Version5370)
+	if shouldPerformUpgrade(sqlStore, Version5360, Version5370) {
```

Comment:
Do we need to provide `hasMissingMigrationsVersion537` to handle partial migrations? That's usually been the pattern with cloud releases.

Reply:
Hmm doesn't the `CRTChannelMembershipCountsMigrationComplete` and `CRTThreadCountsAndUnreadsMigrationComplete` values do this? I think it wasn't needed to check if migration is required if something like these checks are provided.

Reply:
I was actually looking at 5.37 changes. I guess we should be okay considering the version wasn't bumped until now.

Reply:
Ah.. yeah 🤦